### ISSUE TITLE
solve returns empty list for `x**(1/x)  - 1`

### ISSUE THREAD

**Issue by Yathartha22** [sympy/sympy#12538]
At present
```
>>> solve(x**(1/x)  - 1, x) 
[]          --> wrong result                                   
```
should return 

```
>>> solve(x**(1/x)  - 1, x) 
[1]       --> correct output
```

For this equation `solve` returns an empty list, rather it should return `[1]`. I guess there is something that `solve` is missing. I am trying to find out where `solve` is going wrong. I would appreciate if anyone have  suggestions regarding this bug.

@asmeurer @smichr

**Comment by valglad** [comment]
Well, one problem is that `_solve()` complicates the equation when it makes it into a polynomial in `x**(1/x)` on line 1427: it solves the polynomial and proceeds to solve `x**(1/x) - t` so that it can turn the solutions of the polynomial into solutions of the original equation, but of course `x**(1/x) - t` is harder than the original. It would probably be good to have something in that block of code to check for this kind of thing and not allow it. E.g. if the polynomial is linear, then we are only going to make things harder (or are there cases where it would be better after all?) So if you change that, _solve will be able to get to the line where it tries `_tsolve()` and that should give you `[1]`.

**Comment by Yathartha22** [comment]
thanks @valglad
I have also opened a PR regarding this. It would be great if you could give your suggestions regarding the approach

**Comment by oscarbenjamin** [comment]
This is fixed now on master:
```
In [48]: solve(x**(1/x)  - 1, x)                                                                                                                              
Out[48]: [1]
```
I think it was fixed by #15735.

Thanks for the report. I'll close this issue now.
